images/sds-local-volume-csi/pkg/utils/func.go

```diff
@@ -276,20 +296,33 @@ func GetLVMThinPoolFreeSpace(lvg snc.LvmVolumeGroup, thinPoolName string) (thinP
 	return storagePoolThinPool.AvailableSpace, nil
 }
 
-func UpdateLVMLogicalVolume(ctx context.Context, kc client.Client, llv *snc.LVMLogicalVolume) error {
-	var err error
+func ExpandLVMLogicalVolume(ctx context.Context, kc client.Client, llv *snc.LVMLogicalVolume, newSize string) error {
 	for attempt := 0; attempt < KubernetesAPIRequestLimit; attempt++ {
-		err = kc.Update(ctx, llv)
+		llv.Spec.Size = newSize
+		err := kc.Update(ctx, llv)
 		if err == nil {
 			return nil
 		}
-		time.Sleep(KubernetesAPIRequestTimeout)
-	}
 
-	if err != nil {
-		return fmt.Errorf("after %d attempts of updating LVMLogicalVolume %s, last error: %w", KubernetesAPIRequestLimit, llv.Name, err)
+		if attempt < KubernetesAPIRequestLimit-1 {
```

Comment:
Are we sure that we need multiple tries? It's not a k8s's way

Reply:
Fixed